ament_cmake_vendor_package/cmake/ament_vendor.cmake

```diff
@@ -48,6 +48,35 @@
 #   project, expose the external project globally to any downstream CMake
 #   projects.
 # :type GLOBAL_HOOK: option
+# :param IS_VENDORED_OUTPUT_VARIABLE_NAME: the name of the variable that
+#   will be set to ``TRUE`` if the package has been vendored, or ``FALSE``
+#   otherwise.
+# :type IS_VENDORED_OUTPUT_VARIABLE_NAME: string
```

Comment:
Elsewhere in ROS 2 we're already using `if(TARGET target_name)` for this purpose. Is there some reason that approach isn't sufficient?

For example: https://github.com/ros2/libyaml_vendor/blob/c69b99721471d47f88042b44e510cfefafb385b9/CMakeLists.txt#L27

Reply:
Good point! I did not think about this approach, and apparently I am not the only one (all gz vendor packages duplicate the satifisfied condition to check it the package was vendored or not, see https://github.com/gazebo-release/gz_math_vendor/blob/51d2e68e872d2f071bdf40fe7017b9a00ddd8a19/CMakeLists.txt#L67). But using the target defined by the ExternalProject make total sense, I will revise the PR to suggest doing that instead.